CoreScriptsRoot/ServerCoreScripts/ClientChat/ChannelsBar.lua

```diff
@@ -0,0 +1,313 @@
+	LeaveConfirmationNotice.TextStrokeTransparency = 0.75
+	LeaveConfirmationNotice.Text = "Leave channel <XX>?"
+	LeaveConfirmationNotice.Font = Enum.Font.SourceSansBold
+	LeaveConfirmationNotice.FontSize = Enum.FontSize.Size18
+
+	local outPos = LeaveConfirmationFrame.Position
+	LeaveConfirmationButtonYes.MouseButton1Click:connect(function()
+		print("Leave channel")
+		LeaveConfirmationFrame:TweenPosition(outPos, Enum.EasingDirection.Out, Enum.EasingStyle.Quad, 0.2, true)
+	end)
+	LeaveConfirmationButtonNo.MouseButton1Click:connect(function()
+		print("Do not leave channel")
+		LeaveConfirmationFrame:TweenPosition(outPos, Enum.EasingDirection.Out, Enum.EasingStyle.Quad, 0.2, true)
+	end)
+
+
+
+	local scale = 0.7
+	local scaleOther = (1 - scale) / 2
+
+	local PageLeftButton = Instance.new("ImageButton", BaseFrame)
+	PageLeftButton.Name = "PageLeftButton"
+	PageLeftButton.SizeConstraint = Enum.SizeConstraint.RelativeYY
+	PageLeftButton.Size = UDim2.new(scale, 0, scale, 0)
+	PageLeftButton.BackgroundTransparency = 1
+	PageLeftButton.Position = UDim2.new(0, 4, scaleOther, 0)
+	PageLeftButton.Visible = false
+	PageLeftButton.Image = "rbxasset://textures/ui/Chat/TabArrowBackground.png"
+	local ArrowLabel = Instance.new("ImageLabel", PageLeftButton)
+	ArrowLabel.Name = "ArrowLabel"
+	ArrowLabel.BackgroundTransparency = 1
+	ArrowLabel.Size = UDim2.new(0.4, 0, 0.4, 0)
+	ArrowLabel.Image = "rbxasset://textures/ui/Chat/TabArrow.png"
+
+	local PageRightButtonPositionalHelper = Instance.new("Frame", BaseFrame)
+	PageRightButtonPositionalHelper.BackgroundTransparency = 1
+	PageRightButtonPositionalHelper.Name = "PositionalHelper"
+	PageRightButtonPositionalHelper.Size = PageLeftButton.Size
+	PageRightButtonPositionalHelper.SizeConstraint = PageLeftButton.SizeConstraint
+	PageRightButtonPositionalHelper.Position = UDim2.new(1, 0, scaleOther, 0)
+
+	local PageRightButton = PageLeftButton:Clone()
+	PageRightButton.Parent = PageRightButtonPositionalHelper
+	PageRightButton.Name = "PageRightButton"
+	PageRightButton.Size = UDim2.new(1, 0, 1, 0)
+	PageRightButton.SizeConstraint = Enum.SizeConstraint.RelativeXY
+	PageRightButton.Position = UDim2.new(-1, -4, 0, 0)
+
+	local positionOffset = UDim2.new(0.05, 0, 0, 0)
+
+	PageRightButton.ArrowLabel.Position = UDim2.new(0.3, 0, 0.3, 0) + positionOffset
+	PageLeftButton.ArrowLabel.Position = UDim2.new(0.3, 0, 0.3, 0) - positionOffset
+	PageLeftButton.ArrowLabel.Rotation = 180
+
+
+	return BaseFrame, ScrollerFrame, PageLeftButton, PageRightButton, LeaveConfirmationFrame, LeaveConfirmationNotice
+end
+
+
+function methods:UpdateMessagePostedInChannel(channelName)
+	local tab = self:GetChannelTab(channelName)
+	if (tab) then
+		tab:UpdateMessagePostedInChannel()
+	else
+		warn("ChannelsTab '" .. channelName .. "' does not exist!")
+	end
+end
+		
+function methods:AddChannelTab(channelName)
+	if (self:GetChannelTab(channelName)) then
+		error("Channel tab '" .. channelName .. "'already exists!")
```

Comment:
Is it okay for this function to crash? Other calling scripts are protected from potential execution stoppage?